src/naive/mod.rs

```diff
@@ -159,6 +159,34 @@ impl NaiveWeek {
     pub const fn days(&self) -> RangeInclusive<NaiveDate> {
         self.first_day()..=self.last_day()
     }
+
+    /// Returns an [`Option<RangeInclusive<T>>`] representing the whole week bounded by
+    /// [checked_first_day](NaiveWeek::checked_first_day) and
+    /// [checked_last_day](NaiveWeek::checked_last_day) functions.
+    ///
+    /// Returns `None` if either of the boundaries are out of `NaiveDate`'s range
+    /// (more than ca. 262,000 years away from common era).
+    ///
+    ///
+    /// # Examples
+    ///
+    /// todo: examples with min/max
+    /// ```
+    /// use chrono::{NaiveDate, Weekday};
+    ///
+    /// let date = NaiveDate::from_ymd_opt(2022, 4, 18).unwrap();
+    /// let week = date.week(Weekday::Mon);
+    /// let days = week.checked_days().unwrap();
+    /// assert!(days.contains(&date));
+    /// ```
+    #[inline]
+    #[must_use]
+    pub const fn checked_days(&self) -> Option<RangeInclusive<NaiveDate>> {
```

Comment:
I'd expect this to be accompanied by a change elsewhere thats relying on this method except that it panics on `None`, why is that not the case here?

Reply:
I don't understand 🙈
Do you mean adapt some other code within the library to use this new method, reference a code snipped that led to me making this PR, or something else?

Reply:
Changed `days()` as you described (but the downside is that now panic doesn't show which of the boundaries overflowed).
Didn't find usages of panicking functions within the crate for replacement (hopefully `rust-analyzer` searched well 😃 )